src/CommandHandler/RefundUnitsHandler.php

```diff
@@ -18,21 +20,30 @@
     /** @var RefundedUnitTotalProviderInterface */
     private $refundedUnitTotalProvider;
 
+    /** @var OrderRefundingAvailabilityCheckerInterface */
+    private $orderRefundingAvailabilityChecker;
+
     /** @var EventBus */
     private $eventBus;
 
     public function __construct(
         RefundCreatorInterface $refundCreator,
         RefundedUnitTotalProviderInterface $refundedUnitTotalProvider,
+        OrderRefundingAvailabilityCheckerInterface $orderRefundingAvailabilityChecker,
         EventBus $eventBus
     ) {
         $this->refundCreator = $refundCreator;
         $this->refundedUnitTotalProvider = $refundedUnitTotalProvider;
+        $this->orderRefundingAvailabilityChecker = $orderRefundingAvailabilityChecker;
         $this->eventBus = $eventBus;
     }
 
     public function __invoke(RefundUnits $command): void
     {
+        if (!$this->orderRefundingAvailabilityChecker->__invoke($command->orderNumber())) {
+            throw OrderIsNotPaidException::withOrderNumber($command->orderNumber());
```

Comment:
Maybe `OrderNotAvailableForRefunding` or sth similar? In fact we check if order is available for refunding, not if it's paid :)